sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/bitwiseExpressions.scala

```diff
@@ -204,3 +206,51 @@ case class BitwiseCount(child: Expression)
     case LongType => java.lang.Long.bitCount(input.asInstanceOf[Long])
   }
 }
+
+object BitwiseGet {
+  def calculate(target: Int, pos: Int): Int = {
+    val source = target.toBinaryString
```

Comment:
Casting to string seems like a major overhead. It would be better to handle this with bitwise operations.
`(target >> pos) & 1`